Newsletters: add 244 (2023-03-29) (#1059)
bitcoinops/bitcoinops.github.io: _posts/en/newsletters/2023-03-29-newsletter.md

answers posted since our last update.*

{% comment %}<!-- https://bitcoin.stackexchange.com/search?tab=votes&q=created%3a1m..%20is%3aanswer -->{% endcomment %}
{% assign bse = "https://bitcoin.stackexchange.com/a/" %}

FIXME:bitschmidty

## Releases and release candidates

*New releases and release candidates for popular Bitcoin infrastructure
projects.  Please consider upgrading to new releases or helping to test
release candidates.*

- [LND v0.16.0-beta.rc3][] is a release candidate for a new major
  version of this popular LN implementation.

- Rust Bitcoin 0.30.0 FIXME:harding

- BDK 1.0.0 FIXME:harding

<!-- FIXME:harding to update Tuesday -->

## Notable code and documentation changes

*Notable changes this week in [Bitcoin Core][bitcoin core repo], [Core
Lightning][core lightning repo], [Eclair][eclair repo], [LDK][ldk repo],
[LND][lnd repo], [libsecp256k1][libsecp256k1 repo], [Hardware Wallet
Interface (HWI)][hwi repo], [Rust Bitcoin][rust bitcoin repo], [BTCPay
Server][btcpay server repo], [BDK][bdk repo], [Bitcoin Improvement
Proposals (BIPs)][bips repo], and [Lightning BOLTs][bolts repo].*

- [Bitcoin Core #27278][] begins logging by default when a header for a

>   new block is received (unless the node is in Initial Block Download
>   (IBD)).  This was [inspired][obeirne selfish] by multiple node

Reviewer: Maybe avoid the ugly double parenthesis?
```suggestion
  new block is received unless the node is in Initial Block Download
  (IBD).  This was [inspired][obeirne selfish] by multiple node
```

    for spending from the commitment transaction.  It's possible to
    generate private keys in a sequence where anyone who knows one key
    can also derive any later key (but not any earlier key).  This means
    each time Alice revokes a later state, she can give Bob an earlier
    key which Bob can use to derive any later key (for an earlier
    state).  E.g.,

      | Channel state | Key state |
      | 0     | MAX |
      | 1     | MAX - 1 |
      | 2     | MAX - 2 |
      | x     | MAX - x |
      | MAX   | 0 |

    This allows Bob to store all the information he needs to spend from
    an outdated state transaction in a very small constant amount of
    space (less than 100 bytes by our calculations<!-- commitment tx's
    vin outpoint @<=34 bytes, priv key @32 bytes, reference index @<=4
    bytes -->).  The information can also easily be shared with a
    [watchtower][topic watchtowers] (which does not need to be trusted,
    as any successful spending from an outdated state transaction will
    prevent an outdated commitment transaction from being published
    onchain.  Since the funds involved in that outdated state

>     transaction belong entirely to the party who is in breech of the

Reviewer: ```suggestion
    transaction belong entirely to the party who is in breach of the
```

    transactions is also funded by conflicting spends of the funding output
    (meaning that only one of the commitment transactions can eventually be
    confirmed).  With all the child transactions created, the transaction
    which creates the funding output can be signed and broadcast.

    {:.center}
    ![Alice and Bob create their commitment transactions](/img/posts/2023-03-tunable-commitment.dot.png)

    Each of the commitment transactions pays to the current state of the
    channel.  For the initial state (_S<sub>0</sub>_), 50M sats is refunded
    to each Alice and Bob (for simplicity we ignore transaction fees).
    Alice or Bob can start the process of unilaterally closing the
    channel by publishing their version of the state transaction; after
    the enforced delay, they can then publish the corresponding
    commitment transaction.  For example, Alice publishes her state
    transaction and her commitment transaction (which pays both her and
    Bob); at that point, Bob can simply never spend his state
    transaction and instead spend the money used to create it at any
    later time however he would like.

    {:.center}

>     ![Alice spends honestly from the channel](/img/posts/2023-03-tunable-honest-spend.dot.png)

Author: I'm pushing an edit for this as a separate commit in case you don't like it and want to revert before publication (otherwise it can be squashed like normal).

>     ![Alice spends honestly from the channel](/img/posts/2023-03-tunable-honest-spend.dot.png)

Reviewer 2: Dave, thanks I think the details adds clarity (even though also crowding)

    There are two other alternatives to unilaterally closing the channel in
    its initial state.  First, Alice and Bob can cooperatively close the
    channel at any time by spending the funding transaction output (the same
    as done in the current LN protocol).  Second, they could update the

>     state, for example increasing Alice's balance by 10M sat and decreasing

Reviewer 2: ```suggestion
    state, for example, increasing Alice's balance by 10M sat and decreasing
```

    Bob's balance by the same amount.  State _S<sub>1</sub>_ looks similar
    to the initial state (_S<sub>0</sub>_), but to enact it, the previous
    state is revoked by each party giving the other a witness[^keychain] for
    spending the first output from their respective state transactions for
    the previous state (S0).  Neither party can use the other's witness

>     because the S0 state transactions don't themselves contain witnesses yet

Reviewer 2: ```suggestion
    because the _S<sub>0</sub>_ state transactions don't themselves contain witnesses yet
```

    so they can't be broadcast.

    With multiple states available, it's possible to accidentally or
    deliberately close the channel in an outdated state.  For example, Bob may

>     try to close the channel in state S0 where he had an additional 10M

Reviewer 2: ```suggestion
    try to close the channel in state _S<sub>0</sub>_ where he had an additional 10M
```

    satoshis.  To do this, Bob signs and broadcasts his state transaction

>     for S0.  Bob can't take any further action immediately because of

Reviewer 2: ```suggestion
    for _S<sub>0</sub>_.  Bob can't take any further action immediately because of
```

  Pieter Wuille explains the legacy transaction serialization format covered by
  `txid` identifier, the witness extended serialization format covered by the `hash` and
  `wtxid` identifiers, and points out in a [separate answer][se117577] that
  hypothetical additional transaction data would be covered by the `hash` identifier.

- [Can I request tx messages from other peers?]({{bse}}117546)
  User RedGrittyBrick points to resources explaining the [performance][wiki getdata] and
  [privacy][Bitcoin Core #18861] reasons that arbitrary requests for
  transactions from peers is not supported by Bitcoin Core's P2P layer.

- [Eltoo: Does the relative locktime on the first UTXO set the lifetime of the channel?]({{bse}}117468)
  Murch confirms that the question's example [eltoo][topic eltoo]-constructed LN channel
  has a limited lifetime but points to mitigations from the [eltoo
  whitepaper][] that keep the timeouts from expiring.

## Releases and release candidates

*New releases and release candidates for popular Bitcoin infrastructure
projects.  Please consider upgrading to new releases or helping to test
release candidates.*

> - [LND v0.16.0-beta.rc3][] is a release candidate for a new major

Reviewer 2: [lnd v0.16.0-beta.rc5](https://github.com/lightningnetwork/lnd/releases/tag/v0.16.0-beta.rc5)

- [Core Lightning #5986][] updates RPCs which return values in msats to
  no longer include the string "msat" as part of the result.  Instead,
  all returned values are integers.  This completes a deprecation begun
  several releases ago, see [Newsletter #206][news206 msat].

- [Eclair #2616][] adds support for opportunistic [zero-conf
  channels][topic zero-conf channels]---if the remote peer sends the
  `channel_ready` message before the expected number of confirmations,
  Eclair will verify that the funding transaction was entirely created
  by the local node (so the remote peer can't get a conflicting
  transaction confirmed) and then will allow the channel to be used.

- [LDK #2024][] begins including route hints for channels which have
  been opened but which are too immature to have been publicly announced
  yet, such as [zero-conf channels][topic zero-conf channels].

- [Rust Bitcoin #1737][] adds a [security reporting policy][rb sec] for the
  project.

- [BTCPay Server #4608][] allows plugins to expose their features as an
  app in the BTCPay user interface.

- [BIPs #1425][] assigns [BIP93][] to the codex32 scheme for encoding

>   [BIP32][] recovery seeds using Shamir's Secret Sharing (SSS)

Reviewer 2: Not sure that it matter but

```suggestion
  [BIP32][] recovery seeds using Shamir's Secret Sharing Scheme (SSSS)
```

Author: Alliterative maximalism FTW.
